api4/group.go

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api4
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+const (
+	groupMemberActionCreate = iota
+	groupMemberActionDelete
+)
+
+func (api *API) InitGroup() {
+	// GET /api/v4/groups/:group_id
+	api.BaseRoutes.Groups.Handle("/{group_id:[A-Za-z0-9]+}",
+		api.ApiSessionRequiredTrustRequester(getGroup)).Methods("GET")
+
+	// PUT /api/v4/groups/:group_id/patch
+	api.BaseRoutes.Groups.Handle("/{group_id:[A-Za-z0-9]+}/patch",
+		api.ApiSessionRequired(patchGroup)).Methods("PUT")
+
+	for _, syncableType := range model.GroupSyncableTypes {
+		name := strings.ToLower(syncableType.String())
+
+		// POST /api/v4/teams/:team_id/link
+		// POST /api/v4/channels/:channel_id/link
+		api.BaseRoutes.Groups.Handle(fmt.Sprintf("/{group_id:[A-Za-z0-9]+}/%[1]ss/{%[1]s_id:[A-Za-z0-9]+}/link",
```

Comment:
This looks to me like a complex url generation, to avoid duplicate each line. If we don't going to have more "GroupSyncableTypes" in the future, I would simplify this to just, duplicate the lines here.